src/main/scala/Main.scala

```diff
@@ -1,8 +1,84 @@
+
+
+import newmodel.Package
+import newmodel._
+
+import scala.collection.mutable.ListBuffer
+
 /**
  * Created by yaroslav on 11.05.2015.
  */
 object Main {
-  def main(args:Array[String]) ={
-    println("hello world")
+
+  object DummyIndex extends Index{
+    val docElements = ListBuffer.empty[DocElement]
```

Comment:
no mutations please. make DummyIndex a class and let it take an immutable collection as a constructor argument. makes it easier to reason about code. Recommended book: http://www.manning.com/bjarnason/